providers/fips/fipsprov.c

```diff
@@ -542,88 +542,103 @@ int OSSL_provider_init(const OSSL_CORE_HANDLE *handle,
     if (!ossl_prov_seeding_from_dispatch(in))
         return 0;
     for (; in->function_id != 0; in++) {
+        /*
+        * We do not support the scenario of an application linked against
+        * multiple versions of libcrypto (e.g. one static and one dynamic), but
+        * sharing a single fips.so. We do a simple sanity check here.
+        */
+#define set_func(c, f) if (c != NULL && c != f) goto err; else c = f;
```

Comment:
`return 0;` instead of `goto err;` otherwise you would have to do `*provctx = NULL` but that makes no sense as the err block does not do anything useful here.

So I'd write it as: `#define set_func(c, f) if (c == NULL) c = f; else if (c != f) return 0;`